src/server.rs

```diff
@@ -0,0 +1,145 @@
+use std::error::Error;
+use std::sync::Arc;
+
+use tokio::net::UdpSocket;
+use tokio::sync::mpsc::{self, UnboundedReceiver, UnboundedSender};
+use tokio::sync::Mutex;
+use tokio::task::{JoinError, JoinHandle};
+
+use crate::ScuttleButt;
+
+/// Buffer size for UDP messages.
+const BUF_SIZE: usize = 4096;
+
+/// UDP ScuttleButt server.
+pub struct ScuttleServer {
+    pub scuttlebutt: Arc<Mutex<ScuttleButt>>,
+
+    channel: UnboundedSender<ChannelMessage>,
+    join_handle: JoinHandle<()>,
+}
+
+impl ScuttleServer {
+    /// Launch a new server.
+    ///
+    /// This will start the ScuttleButt server as a new Tokio background task.
+    pub fn spawn(node_id: impl Into<String>, port: u16) -> Self {
+        let (tx, rx) = mpsc::unbounded_channel();
+
+        let scuttlebutt = Arc::new(Mutex::new(ScuttleButt::with_node_id(node_id.into())));
+        let scuttlebutt_server = scuttlebutt.clone();
+
+        let join_handle = tokio::spawn(async move {
+            let _ = listen(rx, scuttlebutt_server, port).await;
+        });
+
+        Self {
+            channel: tx,
+            scuttlebutt,
+            join_handle,
+        }
+    }
+
+    /// Perform a ScuttleButt "handshake" with another UDP server.
+    pub fn gossip(&self, addr: impl Into<String>) {
+        let _ = self.channel.send(ChannelMessage::Gossip(addr.into()));
```

Comment:
Can we afford to ignore errors here?

After one kind of trail of events can this return an error and how do we handle that today?

Reply:
>  After one kind of trail of events can this return an error and how do we handle that today?

I'm not sure what you mean by "trail of events" here. This should only return an error when the server has shut down and the receiver was dropped. It's probably useful to return a Result from it just in case.